classes/target/target_dataroot.php

```diff
@@ -251,4 +262,31 @@ public function is_available() {
         return false;
     }
 
+    /**
+     * Returns full path of target file.
+     *
+     * @param string $file temp file path or filename.
+     *
+     * @return string
+     */
+    protected function get_target_file_path($file) {
+        return rtrim($this->get_full_path(), '/') . '/' . $this->get_target_file_name($file);
+    }
+
+    /**
+     * Returns array of available date formats.
+     *
+     * @return array
+     */
+    protected function get_list_of_formats() {
+
+        $availableformats = array('dmY_hi', 'Ydm_hi');
+        $formats = array();
+
+        foreach ($availableformats as $format) {
+            $formats[$format] = $format;
```

Comment:
Would be good to actually display a date in the list of options like:
$formats[$format] = date($format, time());